include/fastcdr/Cdr.h

```diff
@@ -2761,7 +2761,7 @@ class Cdr
         if (!value)
         {
             value = external<_T>{new
```

Comment:
We could just use `value = external<_T>{new typename external<_T>::type()};` for all compilers. Note that `typename` is required in this case.

It might be even better to add `typename` to the `using` clause in `external`, but I haven't looked at how `external<T>::type` is being used in other parts of the code:

```c++
template<class T>
class external
{
public:
    using type = typename T;
...
};
```